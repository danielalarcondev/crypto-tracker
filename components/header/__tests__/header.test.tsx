import * as React from 'react';
import '@testing-library/jest-dom/jest-globals';

import { render, screen } from '@testing-library/react';
import { expect, it, describe } from '@jest/globals';
import Header from '@components/header/header';

const MOCKED_PATHNAME = 'https://test.com/assets';

jest.mock('next/navigation', () => ({
    usePathname: () => MOCKED_PATHNAME
}));

Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query: string) => {

        if (query === '(prefers-color-scheme: dark)') {
            return {
                media: '(prefers-color-scheme: light)',
                matches: false,
            };
        }

        return {
            media: '(prefers-color-scheme: light)',
            matches: true,
        };

    })
});

describe('Header: ', () => {

    it('should render correctly', () => {
        const { container } = render(<Header />);

        const ul = screen.getByTestId('header-ul');
        const logo = screen.getByTestId('header-logo');
        const settings = screen.getByTestId('header-settings-icon');

        expect(ul).toBeInTheDocument();
        expect(logo).toBeInTheDocument();
        expect(settings).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
