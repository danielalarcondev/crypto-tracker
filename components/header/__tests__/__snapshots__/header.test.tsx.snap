// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header:  should render correctly 1`] = `
<div>
  <header
    class="header max-w-[3840px] fixed top-0 w-full z-20 border-b flex shadow-[0px_10px_15px_-10px_rgba(0,0,0,0.3)] border-neutral-300 dark:border-neutral-700 bg-white dark:bg-neutral-800 h-16"
  >
    <nav
      class="w-full max-w-[1920px] my-0 mx-auto"
      data-testid="header-nav"
    >
      <div
        class="flex flex-1 w-full md:justify-between pl-2 md:pl-12 gap-6 md:gap-12 h-full flex-wrap"
      >
        <div
          class="content-center sm:hidden"
        >
          <svg
            class="w-8 h-8"
            data-testid="header-menu-button"
            fill="none"
            stroke="currentColor"
            stroke-width="1.5"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.75 6.75h16.5M3.75 12h16.5m-16.5 5.25h16.5"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </div>
        <div
          class="flex flex-wrap pt-2 pr-2 h-full flex-1 sm:flex-[none] justify-end md:justify-normal"
          data-testid="header-logo"
        >
          <a
            href="/tracker"
          >
            <img
              alt="logo"
              class="w-12 h-12"
              data-nimg="1"
              decoding="async"
              height="48"
              loading="lazy"
              src="/logo.svg"
              style="color: transparent;"
              width="48"
            />
          </a>
        </div>
        <div
          class="flex-1 w-full md:w-auto h-full md:items-center md:order-1 hidden sm:flex"
          id="navbar-sticky"
        >
          <ul
            class="menu-ul flex flex-1 font-medium rounded-lg bg-transparent justify-start h-full"
            data-testid="header-ul"
          >
            <li
              class="pl-3 pr-3 text-xl hover:cursor-pointer border-b-4 hover:text-primary-500 border-transparent w-fit h-[calc(100%+1px)]"
            >
              <a
                class="menu-anchor block md:bg-transparent h-full flex items-center mt-1"
                href="/tracker"
              >
                Tracker
              </a>
            </li>
            <li
              class="pl-3 pr-3 text-xl hover:cursor-pointer border-b-4 hover:text-primary-500 border-transparent w-fit h-[calc(100%+1px)]"
            >
              <a
                class="menu-anchor block md:bg-transparent h-full flex items-center mt-1"
                href="/assets"
              >
                Assets
              </a>
            </li>
          </ul>
        </div>
        <div
          class="w-12 md:w-24 cursor-pointer order-2 items-center h-full hidden sm:inline-flex"
        >
          <button
            class="absolute w-fit h-fit"
            data-testid="header-settings-icon"
          >
            <svg
              class="w-8 h-8 fill-none hover:transform hover:rotate-12 hover:stroke-primary-500 stroke-black dark:stroke-white"
              stroke-width="1.5"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.594 3.94c.09-.542.56-.94 1.11-.94h2.593c.55 0 1.02.398 1.11.94l.213 1.281c.063.374.313.686.645.87.074.04.147.083.22.127.325.196.72.257 1.075.124l1.217-.456a1.125 1.125 0 0 1 1.37.49l1.296 2.247a1.125 1.125 0 0 1-.26 1.431l-1.003.827c-.293.241-.438.613-.43.992a7.723 7.723 0 0 1 0 .255c-.008.378.137.75.43.991l1.004.827c.424.35.534.955.26 1.43l-1.298 2.247a1.125 1.125 0 0 1-1.369.491l-1.217-.456c-.355-.133-.75-.072-1.076.124a6.47 6.47 0 0 1-.22.128c-.331.183-.581.495-.644.869l-.213 1.281c-.09.543-.56.94-1.11.94h-2.594c-.55 0-1.019-.398-1.11-.94l-.213-1.281c-.062-.374-.312-.686-.644-.87a6.52 6.52 0 0 1-.22-.127c-.325-.196-.72-.257-1.076-.124l-1.217.456a1.125 1.125 0 0 1-1.369-.49l-1.297-2.247a1.125 1.125 0 0 1 .26-1.431l1.004-.827c.292-.24.437-.613.43-.991a6.932 6.932 0 0 1 0-.255c.007-.38-.138-.751-.43-.992l-1.004-.827a1.125 1.125 0 0 1-.26-1.43l1.297-2.247a1.125 1.125 0 0 1 1.37-.491l1.216.456c.356.133.751.072 1.076-.124.072-.044.146-.086.22-.128.332-.183.582-.495.644-.869l.214-1.28Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
              <path
                d="M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </button>
          <div
            class="relative w-40 h-16 -bottom-[55px] left-[-105px] shadow-md rounded-b outline outline-1 -outline-offset-1 outline-primary-500 bg-white dark:bg-neutral-800 hidden"
            data-testid="header-settings-overlay"
          >
            <div
              class="w-36 h-16 p-2 flex justify-center flex-wrap"
            >
              <label
                class="inline-flex items-center cursor-pointer w-auto order-2"
              >
                <div
                  class="mr-2"
                >
                  <svg
                    class="w-6 h-6 stroke-black dark:stroke-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M12 3v2.25m6.364.386-1.591 1.591M21 12h-2.25m-.386 6.364-1.591-1.591M12 18.75V21m-4.773-4.227-1.591 1.591M5.25 12H3m4.227-4.773L5.636 5.636M15.75 12a3.75 3.75 0 1 1-7.5 0 3.75 3.75 0 0 1 7.5 0Z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
                <input
                  class="theme-switcher sr-only peer"
                  data-testid="header-theme-toggle-input"
                  type="checkbox"
                />
                <div
                  class="relative w-11 after:w-5 h-6 after:h-5 bg-gray-200 peer-checked:bg-primary-500 rounded-full peer-checked:after:translate-x-full rtl:peer-checked:after:-translate-x-full peer-checked:after:border-white after:content-[""] after:absolute after:top-0.5 after:start-0.5 after:bg-white after:border-gray-300 after:border after:rounded-full after:transition-all"
                />
                <div
                  class="ml-2"
                >
                  <svg
                    class="w-6 h-6 stroke-black dark:stroke-white"
                    fill="none"
                    stroke="currentColor"
                    stroke-width="1.5"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M21.752 15.002A9.72 9.72 0 0 1 18 15.75c-5.385 0-9.75-4.365-9.75-9.75 0-1.33.266-2.597.748-3.752A9.753 9.753 0 0 0 3 11.25C3 16.635 7.365 21 12.75 21a9.753 9.753 0 0 0 9.002-5.998Z"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                    />
                  </svg>
                </div>
              </label>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
